curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -297,31 +297,44 @@ try {
 }
 ```
 
-When the input array is not empty, the function `getRandomLunch` should log a string in the format `"Randomly selected lunch: [Lunch Item]"` to the console.
+When the input array is not empty, the function `getRandomLunch` should log a string in the format `"Randomly selected lunch: [Lunch Item]"` to the console. Additionally, your function `getRandomLunch` should not modify the input array.
 
 ```js
 const testLunches = ["Fish", "Fries", "Roast"];
 const tempRandom = Math.random;
 const tempArr = [];
 const temp = console.log;
 
+// Helper function that returns whether the input array to a function was modified
+function checkIfArrayIsModified(original, callback) {
+  const arrayCopy = [...original];
+  callback(original);
+  return !(original.length === arrayCopy.length && original.every((value, index) => value === arrayCopy[index]));
+}
+
 try {
+  let isModified; 
+  const regex = /Randomly selected lunch: \w+/  // Output format to compare results with
+
   console.log = obj => tempArr.push(obj);
 
-  // check that it correctly outputs the first item
+  // check that it correctly outputs the first item and that the original array was not modified by getRandomLunch()
```

Comment:
you should create a different test that specifically tests for the non-mutation of the array, adding it to this test here seems too much stuff

having two different tests make it more clear if it's failing because the logged string is wrong, or because the array is mutated